lib/libc/common/source/thrd/mtx.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <threads.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/posix/pthread.h>
+
+int mtx_init(mtx_t *mutex, int type)
+{
+	int ret;
+	pthread_mutexattr_t attr;
+	pthread_mutexattr_t *attrp = NULL;
+
+	switch (type) {
+	case mtx_plain:
+	case mtx_timed:
+		break;
+	case mtx_plain | mtx_recursive:
+	case mtx_timed | mtx_recursive:
+		attrp = &attr;
+		ret = pthread_mutexattr_init(attrp);
+		__ASSERT_NO_MSG(ret == 0);
+
+		ret = pthread_mutexattr_settype(attrp, PTHREAD_MUTEX_RECURSIVE);
+		__ASSERT_NO_MSG(ret == 0);
+		break;
+	default:
+		return thrd_error;
+	}
+
+	switch (pthread_mutex_init(mutex, attrp)) {
```

Comment:
for 2 results, is it really worth having a switch or just a simple if/else?  I won't repeat the question on the other places where this happens as well.

Not really a requesting a change, just a minor question on code intent.  A `switch` implies that more conditional results will be of value in the future, but any of the other possible results for this don't matter, they are just failures.

Reply:
> A `switch` implies that more conditional results will be of value in the future, but any of the other possible results for this don't matter, they are just failures.

Yes, that's exactly it. I wanted to structure it so that errors could easily be more specialized or be corrected with ease.

Since you were curious though, here is a comparison 
https://godbolt.org/z/GzPv4nf5c

with default optimizations
```diff
diff -Nua /tmp/*.S
--- /tmp/with-if-stmt.S	2023-09-07 19:52:25
+++ /tmp/with-switch-stmt.S	2023-09-07 19:52:44
@@ -10,11 +10,11 @@
         ldr     x0, [sp, 40]
         bl      pthread_mutex_init
         cmp     w0, 0
-        beq     .L2
-        mov     w0, 2
+        bne     .L2
+        mov     w0, 0
         b       .L3
 .L2:
-        mov     w0, 0
+        mov     w0, 2
 .L3:
         ldp     x29, x30, [sp], 48
         ret

```

with `-O3`
```diff
<intentionally empty>
```